test/integration/node-specific/abort_signal.test.ts

```diff
@@ -0,0 +1,872 @@
+              });
+            }
+
+            for (const [cursorAPI, { value: args }] of getAllProps(cursorAPIs)) {
+              test(cursorAPI, args);
+            }
+          });
+        });
+      }
+    });
+  }
+
+  testCursor('find', FindCursor);
+  testCursor('aggregate', AggregationCursor);
+  testCursor('listCollections', ListCollectionsCursor);
+
+  describe('cursor stream example', () => {
+    beforeEach(async function () {
+      const utilClient = this.configuration.newClient();
+      try {
+        const collection = utilClient.db('abortSignal').collection('support');
+        await collection.drop({}).catch(() => null);
+        await collection.insertMany([
+          { a: 1, ssn: '0000-00-0001' },
+          { a: 2, ssn: '0000-00-0002' },
+          { a: 3, ssn: '0000-00-0003' }
+        ]);
+      } finally {
+        await utilClient.close();
+      }
+    });
+
+    it('follows expected stream error handling', async () => {
+      const controller = new AbortController();
+      const { signal } = controller;
+      const cursor = collection.find({}, { signal, batchSize: 1 });
+      const cursorStream = cursor.stream();
+
+      const { promise, resolve, reject } = promiseWithResolvers<void>();
+
+      cursorStream
+        .on('data', () => controller.abort())
+        .on('error', reject)
+        .on('close', resolve);
+
+      expect(await promise.catch(error => error)).to.be.instanceOf(DOMException);
+    });
+  });
+
+  describe('KMS requests', function () {
+    const stateMachine = new StateMachine({} as any);
+    const request = {
+      addResponse: _response => undefined,
+      status: {
+        type: 1,
+        code: 1,
+        message: 'notARealStatus'
+      },
+      bytesNeeded: 500,
+      kmsProvider: 'notRealAgain',
+      endpoint: 'fake',
+      message: Buffer.from('foobar')
+    };
+
+    let controller: AbortController;
+    let signal: AbortSignal;
+    let cursor: AbstractCursor<{ a: number }>;
+
+    beforeEach(async function () {
+      controller = new AbortController();
+      signal = controller.signal;
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+      await cursor?.close();
+    });
+
+    describe('when StateMachine.kmsRequest() is passed an AbortSignal', function () {
+      beforeEach(async function () {
+        sinon.stub(TLSSocket.prototype, 'connect').callsFake(function (..._args) {
+          return this;
+        });
+      });
+
+      afterEach(async function () {
+        sinon.restore();
+      });
+
+      it('the kms request rejects when signal is aborted', async function () {
+        const err = stateMachine.kmsRequest(request, { signal }).catch(e => e);
+        await sleep(1);
+        controller.abort();
+        expect(await err).to.be.instanceOf(DOMException);
+      });
+    });
+  });
+
+  describe('when a signal passed to countDocuments() is aborted', failPointMetadata, () => {
```

Comment:
Why count documents, when estimatedDocumentCount + other non-find/aggregate cursors don't get signal support?

Reply:
Specifically because countDocuments is just an aggregation wrapper, it got the same treatment findOne did. I understand it should be an easy (maybe even TS only) change to add support to those but each one is just a little more scope and a little more time.

Reply:
I'm coming at this from the perspective of a user - when this feature is published, users will be confused why it applies to countDocuments but not other APIs.  Users likely have no knowledge of the internals of our count APIs.

I'd suggest either expanding support to all relevant APIs, or just removing support from countDocuments.

Reply:
To mitigate confusion, I have the release notes point out "we're starting small", "partial support", and I list the exact APIs added. Our typescript also makes it clear by only adding the support to the APIs that accept it. I don't think removing `countDocuments` helps with clarity. 

If you are unaware of the aggregation supporting `countDocuments`, you're happy to see signal supported and wouldn't think much more of it, if you are aware, you understand why: because it is a pass-through option.

If you are using an API that didn't get added here, there's a handful of communication avenues to let us know and we'll plan the effort (or accept a contribution)